astropy/time/tests/test_precision.py

```diff
@@ -140,33 +266,221 @@ def test_leap_seconds_rounded_correctly():
+    t1 = Time(mjd+1, format="mjd", scale="utc")
+    assert_quantity_allclose((t1-t0).to(u.s), (1*u.day+delta*u.s))
+    assert_quantity_allclose((th-t0).to(u.s), f*(1*u.day+delta*u.s))
+    assert_quantity_allclose((t1-th).to(u.s), (1-f)*(1*u.day+delta*u.s))
+
+
+@given(sampled_from(_scales), unreasonable_jd(), floats(-10000, 10000))
+@example(scale='utc',
+         jds=(0.0, 2.2204460492503136e-13),
+         delta=6.661338147750941e-13)
+@example(scale='utc',
+         jds=(2441682.5, 2.2204460492503136e-16),
+         delta=7.327471962526035e-12)
+def test_jd_add_subtract_round_trip(scale, jds, delta):
+    jd1, jd2 = jds
+    if scale == 'utc' and 2400000 > jd1+jd2:
+        # Old UTC times degrade accuracy
+        thresh = 100*u.ns
+    else:
+        thresh = 2*dt_tiny
+    t = Time(jd1, jd2, scale=scale, format="jd")
+    try:
+        t2 = t + delta*u.day
+        if abs(delta) >= np.finfo(float).eps:
+            assert t2 != t
+        t3 = t2 - delta*u.day
+        assert_almost_equal(t3, t, atol=thresh, rtol=0)
+    except ErfaError:
+        assume(scale != 'utc' or 2440000 < jd1+jd2 < 2460000)
+        raise
+
+
+@given(sampled_from(_scales), unreasonable_jd(), unreasonable_jd())
+@example(scale='utc',
+         jds_a=(2455000.0, 0.0),
+         jds_b=(2443144.5, 0.5000462962962965))
+def test_timedelta_full_precision(scale, jds_a, jds_b):
+    jd1_a, jd2_a = jds_a
+    jd1_b, jd2_b = jds_b
+    assume(scale != 'utc'
+           or (2440000 < jd1_a+jd2_a < 2460000
+               and 2440000 < jd1_b+jd2_b < 2460000))
+    t_a = Time(jd1_a, jd2_a, scale=scale, format="jd")
+    t_b = Time(jd1_b, jd2_b, scale=scale, format="jd")
+    dt = t_b - t_a
+    assert dt != (t_b + dt_tiny) - t_a
+    assert_almost_equal(t_b-dt/2, t_a+dt/2, atol=2*dt_tiny, rtol=0)
+    assert_almost_equal(t_b+dt, t_a+2*dt, atol=2*dt_tiny, rtol=0)
+    assert_almost_equal(t_b-2*dt, t_a-dt, atol=2*dt_tiny, rtol=0)
+
+
+@given(sampled_from(_scales),
+       unreasonable_jd(),
+       unreasonable_jd(),
+       integers(1, 100),
+       integers(1, 100))
+def test_timedelta_full_precision_arithmetic(scale, jds_a, jds_b, x, y):
+    jd1_a, jd2_a = jds_a
+    jd1_b, jd2_b = jds_b
+    t_a = Time(jd1_a, jd2_a, scale=scale, format="jd")
+    t_b = Time(jd1_b, jd2_b, scale=scale, format="jd")
+    try:
+        dt = t_b - t_a
+        dt_x = x*dt/(x+y)
+        dt_y = y*dt/(x+y)
+        assert_almost_equal(
+            dt_x + dt_y, dt, atol=(x+y)*dt_tiny, rtol=0)
+    except ErfaError:
+        assume(scale != 'utc'
+               or (2440000 < jd1_a+jd2_a < 2460000
+                   and 2440000 < jd1_b+jd2_b < 2460000))
+        raise
+
+
+@given(sampled_from(_scales), sampled_from(_scales),
+       reasonable_jd(), reasonable_jd())
+def test_timedelta_conversion(scale1, scale2, jds_a, jds_b):
+    jd1_a, jd2_a = jds_a
+    jd1_b, jd2_b = jds_b
+    # not translation invariant so can't convert TimeDelta
+    assume(scale1 != 'utc')
+    assume(scale1 != 'ut1')
```

Comment:
UT1 and UT1 should be OK (counted in earth rotation cycles, so continuous). But I guess you really are aiming for different scale1 and scale2 here - in that case, maybe make the test explicitly do that? (skip it if scales are equal?)

Reply:
Well it surprises me a little that UT1-UT1 might work so I changed the assumptions to permit that (UTC-UTC does not).

Reply:
We had quite a bit of discussion about it at the time, but came to the conclusion that since UT1 is continuous, it should work. It mostly helps in that you can, therefore, just adds quantities in units of time to it (which have indeterminate scale, represented with `scale=None` in `TimeDelta`).